### PR TITLE
Allow using any window as wxStaticBox label

### PR DESCRIPTION
**Pull request by vadz**

### CODE REVIEW

File: interface/wx/statbox.h [hunk 1]
```diff
@@ -97,5 +137,51 @@ class wxStaticBox : public wxControl
...
+
+        See the constructor documentation for more details.
+
+        Currently this overload is only available in wxGTK, use
+        @c wxHAS_WINDOW_LABEL_IN_STATIC_BOX to check whether it can be used at
+        compile-time.
+
+        @since 3.1.1
+     */
+    wxStaticBox(wxWindow* parent, wxWindowID id,
```
Comment (MaartenBent) [inline interface/wx/statbox.h]: I guess this should be `Create()` instead of the constructor?
And its also available for `wxMSW`.

**Comment by MaartenBent** [comment]
Seems to build and run fine on Windows with msvc and mingw. Also with a `wxChoice` as label.

It doesn't seem to underline accelerators anymore? E.g. in the widgets sample on the `Spin` page it used to underline the S of `Set style` and the T of `Text alignment`.

File: interface/wx/statbox.h (hunk 1 above)
Comment (vadz) [reply interface/wx/statbox.h]: Fixed, thanks!

**Comment by vadz** [comment]
Thanks for reviewing and testing! I don't know what has happened with the labels, they shouldn't have been affected by these changes.

I did find another nasty bug with the box enabled state in the meanwhile, which I fixed now (hopefully without breaking anything else).

One related remaining problem is that, as I've noticed only recently, MSW doesn't actually grey out the labels of disabled group boxes. I wonder if we should do it manually because they just don't look right otherwise.

**Pull request merged**